sdk/communication/azure-communication-identity/azure/communication/identity/_communication_identity_client.py

```diff
@@ -78,75 +85,73 @@ def from_connection_string(
 
     @distributed_trace
     def create_user(self, **kwargs):
-        # type: (...) -> CommunicationUserIdentifier
+        # type: (Any) -> CommunicationIdentifier
         """create a single Communication user
 
-        :return: CommunicationUserIdentifier
-        :rtype: ~azure.communication.identity.CommunicationUserIdentifier
+        :return: CommunicationIdentifier
+        :rtype: ~azure.communication.identity.CommunicationIdentifier
         """
         return self._identity_service_client.communication_identity.create(
-            cls=lambda pr, u, e: CommunicationUserIdentifier(u.identity.id),
+            cls=lambda pr, u, e: CommunicationUserIdentifier(u.identity.id, raw_id=u.identity.id),
             **kwargs)
 
     @distributed_trace
     def create_user_and_token(
             self,
-            scopes, # type: List[Union[str, "_model.CommunicationTokenScope"]]
+            scopes, # type: List[Union[str, CommunicationTokenScope]]
             **kwargs # type: Any
         ):
-        # type: (...) -> Tuple[CommunicationUserIdentifier, AccessToken]
-        """create a single Communication user with an identity token.
-        :param scopes:
-            List of scopes to be added to the token.
-        :type scopes: list[str or ~azure.communication.identity.models.CommunicationTokenScope]
-        :return: A tuple of a CommunicationUserIdentifier and a AccessToken.
+        # type: (...) -> Tuple[CommunicationIdentifier, AccessToken]
+        """Create a single Communication user with an identity token.
+
+        :param scopes: List of scopes to be added to the token.
+        :type scopes: list[str or ~azure.communication.identity.CommunicationTokenScope]
+        :return: A tuple of a CommunicationIdentifier and a AccessToken.
         :rtype:
-            tuple of (~azure.communication.identity.CommunicationUserIdentifier, ~azure.core.credentials.AccessToken)
+            tuple of (~azure.communication.identity.CommunicationIdentifier, ~azure.core.credentials.AccessToken)
         """
         return self._identity_service_client.communication_identity.create(
-            cls=lambda pr, u, e: (CommunicationUserIdentifier(u.identity.id),
+            cls=lambda pr, u, e: (CommunicationUserIdentifier(u.identity.id, raw_id=u.identity.id),
                 AccessToken(u.access_token.token, u.access_token.expires_on)),
             create_token_with_scopes=scopes,
             **kwargs)
 
     @distributed_trace
     def delete_user(
             self,
-            communication_user, # type: CommunicationUserIdentifier
+            user, # type: CommunicationUserIdentifier
```

Comment:
Now that we're only generalizing the create return type, will this prevent me from calling `create_user` and then passing the returned value into `delete_user` or `get_token`?

Reply:
Nope - it would be non-breaking to expand an input to accept multiple types, but it _would_ be breaking to expand an output type to be multiple types. So we can update the input type if these APIs evolve in future.

Reply:
That makes sense. What I was wondering though is why I can pass a `CommunicationIdentifier` protocol for a `CommunicationUserIdentifier` argument. Isn't the user type narrower than the identifier protocol and would reject it?

Reply:
Right now you cannot pass a CommunicationIdentifier protocol as an argument - the implementation is written to only work with CommunicationUserIdentifiers. So currently the input type is accurate - only that specific type is supported. I'm assuming that if the service starts supporting other Identifier types in this library it will require an SDK release - to support that, at which point we would evaluate the type hints.

As an aside - none of this prevents runtime usage. These typehints are just there for the IDE experience and static analysis.